dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datasetutility/OptionalFileParams.java

```diff
@@ -256,7 +264,7 @@ private void msgt(String s) {
     /**
      * Add parameters to a DataFile object
      */
-    public void addOptionalParams(DataFile df) throws DataFileTagException {
```

Comment:
I was thinking how we can avoid passing those beans, aside from that the beans shouldn't be in the Util classes but they made it that way. In theory this class also should be bean. So i think you can try to inject the licenseDAO and factory in this class. Then you wouldn't have to drag those classes all the way here.

Reply:
I did try it. I've decided that rather than trying to make a `Stateful` service out of it, passing parameters like that is still a better solution.

Reply:
Why do you think it is better? If this supposed to be a class without container it shouldn't contain any other bean that needs to be injected, otherwise it needs to be a service since it becomes a hybrid where some methods you can use normally and for some you need to inject the beans before you can even use the methods. So it needs to be one way or the other. So we find some way to not pass the beans or we make it a service where we inject the beans we want.

Reply:
The way this whole processing is being made, I don't see how this can be accomplished. We can discuss this during sprint review.